File: commands/music/skip.js
/* global controller guild */

module.exports = {
  name: 'skip',
  description: 'Skippar den nuvarande låten',
  aliases: [ 'skippa', 'byt', 'sk' ],
  group: 'music',
  usage: 'skip',
  example: 'skip',
  serverOnly: true,
  adminOnly: false,
  async execute( message, args ) {
    if ( guild.voiceConnection && controller.queue.length > 0 ) {
      controller.skip();
      if ( controller.queue.length === 0 ) return message.reply( 'Kön är slut.' ).then( msg => { msg.delete( 10000 ); } );
      else {
        message.reply( 'skippar låten' ).then( msg => { msg.delete( 10000 ); } );
      }
    } else return message.reply( 'kan inte skippa en låt då boten inte spelar någon musik.' ).then( msg => { msg.delete( 10000 ); } );
  },
};

File: commands/music/skip.go
package music

import (
	"github.com/digitalungdom-se/dub/internal"
	"github.com/digitalungdom-se/dub/pkg"
)

var Skip = internal.Command{
	Name:        "skip",
	Description: "Skippar den nuvarande låten",
	Aliases:     []string{"skippa", "byt", "sk"},
	Group:       "music",
	Usage:       "skip",
	Example:     "skip",
	ServerOnly:  true,
	AdminOnly:   false,

	Execute: func(ctx *pkg.Context, server *internal.Server) error {
		ctx.Delete()
		server.Controller.Skip()

		return nil
	},
}
